src/main/java/smartthings/ratpack/aws/AwsModule.java

```diff
@@ -0,0 +1,41 @@
+package smartthings.ratpack.aws;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import ratpack.guice.ConfigurableModule;
+import smartthings.ratpack.aws.internal.providers.DefaultAWSCredentialsProvider;
+
+/**
+ * A base module for configuring communication with AWS.
+ */
+public class AwsModule extends ConfigurableModule<AwsModule.Config> {
+
+    @Override
+    protected void configure() {
+        bind(AWSCredentialsProvider.class).toProvider(DefaultAWSCredentialsProvider.class);
+    }
+
+    /**
+     * AwsModule Config.   `awsSecretKey` and `awsAccessKey` are optional.  When not  present module will attempt
+     * to resolve via environment vars or instance profile.
+     */
+    public static class Config {
```

Comment:
What about roleArn?  To make this useful we at least need to include the ability to use STS.

Reply:
Don't know if I fully agree on it being unusable w/o STS support.  However, I agree it is nice to have.  I added support directly in the `DefaultAWSCredentialsProvider` class.  Also, this provider is wired as a Guice optional binding, so library consumers are free to override provider binding and have any credentials logic they want.